Sources/Views/AnimatedImageView.swift

```diff
@@ -488,18 +496,21 @@ extension AnimatedImageView {
         }
 
         private func loadFrame(at index: Int) -> UIImage? {
-            guard let image = CGImageSourceCreateImageAtIndex(imageSource, index, nil) else {
+            let options: [CFString: Any] = [
+                kCGImageSourceCreateThumbnailFromImageIfAbsent: true,
+                kCGImageSourceCreateThumbnailWithTransform: true,
+                kCGImageSourceShouldCacheImmediately: true,
+                kCGImageSourceThumbnailMaxPixelSize: max(size.width, size.height)
+            ]
+
+            guard let cgImage = CGImageSourceCreateImageAtIndex(imageSource,
```

Comment:
I guess this would trigger a trap if the size is `.zero`. Maybe we should follow the existing logic to ignoring the `needsPrescaling` when `.zero` is given.